### PR TITLE
Add conditional throttling howto page

### PR DESCRIPTION
**Pull request by praminda**
### Purpose

- Add a how to page for conditional throttling.
- Add a overview page for rate limiting
### Issues

Fixes #

---
#### Maintainers: Check before merge
- [ ] Manual verification

### CODE REVIEW

**Comment by VirajSalaka** [comment]
We need to mention the pilot node  in the overview right, when it comes to application throttling and subscription throttling.

**Comment by praminda** [comment]
> We need to mention the eventhub in the overview right, when it comes to application throttling and subscription throttling.

Yes we need to. Didn't add anything new here since you are anyway going to change the page. Just moved the overview page and pointed to howtos.

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 1]
```diff
@@ -1,51 +1,36 @@
...
+
+The API Microgateway upon recieving a request, checks against the local counter and if throttling limit  has not exceeded it publishes the events via a stream to a central traffic management solution. This is done over HTTP. The  central traffic management solution then  executes throttle policies against the events streams. When a particular request is throttled, the  central traffic management solution sends the details of the throttled out event to a JMS topic. Each API Microgateway node is subscribed to this JMS topic, and updates the local counter when the JMS topic is updated. Hence the API Microgateway nodes gets notified of the throttle decisions through JMS messages.
+
+![distributed throttling](/assets/img/how-tos/distributed-throttling.png)
 
 ### Enabling distributed throttling
 
-1.  Let's create a microgateway project.
+1.  Lets create a microgateway project
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:11]:
```suggestion
1.  Let's create a microgateway project
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 2]
```diff
@@ -1,51 +1,36 @@
...
+
+    ``` java tab="Example"
+    micro-gw init petstore
+    ```
+
+    ```text tab="Response"
+    Project 'petstore' is initialized successfully.
+    ```
+
+2.  Add the API definition(s) to `petstore/api_definitions` directory. A sample open API definition can be found [here](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml). Then provide the API throttling tier using following extension at the API level.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:27]:
```suggestion
2.  Add the API definition(s) to `petstore/api_definitions` directory. A sample open API definition can be found [here](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml). Then provide the API throttling tier using the following extension at the API level.
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 3]
```diff
@@ -162,18 +152,18 @@ and perform rate-limiting precisely.
     
 9.  Finally the added configurations in `microgw.conf` would look like this.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:153]:
```suggestion
9.  Finally, the added configurations in `microgw.conf` would look like this. 
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 4]
```diff
@@ -186,10 +176,58 @@ and perform rate-limiting precisely.
...
     ```
 
-    ``` java tab="Example"
+    ```java tab="Example"
     gateway /Users/kim/petstore/target/petstore.jar
     ```
+
+### Conditional throttling
+
+There can be situations where a certain APIs require more granular level of throttling. Assume you want to provide limited access to a certain IP range or a type of client applications (identified by User-Agent header). For these scenarios, a simple throttle policy with API/resource level limits is not sufficient. To address complex throttling requirements as above, microgateway is capable of throttling request based on several conditions. Following types of conditions are supported.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:189]:
```suggestion
There can be situations where certain APIs require more granular level of throttling. Assume you want to provide limited access to a certain IP range or a type of client application (identified by User-Agent header). For these scenarios, a simple throttle policy with API/resource level limits is not sufficient. To address complex throttling requirements as above, microgateway is capable of throttling requests based on several conditions. The following types of conditions are supported.
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 5]
```diff
@@ -186,10 +176,58 @@ and perform rate-limiting precisely.
...
+    This condition can be used to provide specific limits to a certain IP address or a range of IP addresses.
+
+1.  Header conditions.      
+    This condition can be used to set specific limits to a certain header value.
+
+1.  Query parameter conditions.     
+    Same as the header conditions, this allows applying a specific limit to a certain query parameter value.
+
+1.  JWT claim conditions.       
+    This type of conditions will evaluate the [backend jwt](/how-tos/passing-enduser-attributes-to-the-backend-using-jwt) and check if it has a specific claim value in it to set the throttle limit.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:201]:
```suggestion
    This type of condition will evaluate the [backend jwt](/how-tos/passing-enduser-attributes-to-the-backend-using-jwt) and check if it has a specific claim value in it to set the throttle limit.
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 6]
```diff
@@ -186,10 +176,58 @@ and perform rate-limiting precisely.
...
+
+#### Configure and enable conditional throttling
+
+1.  Open `micro-gw.conf` file in `<MGW_HOME>/conf` directory.
+1.  Add/Enable below configurations to enable each of the required condition type.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:206]:
```suggestion
1.  Add/Enable below configurations to enable the required condition type.
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 7]
```diff
@@ -186,10 +176,58 @@ and perform rate-limiting precisely.
...
+    ```toml
+    [throttlingConfig]
+      enabledGlobalTMEventPublishing = true
+      enableHeaderConditions = true
+      enableQueryParamConditions = true
+      enableJwtClaimConditions = true
+    ```    
+
+1.  Define the Advance Throttle Policy containing the required conditions in WSO2 API Manager. To do this follow [Adding a new advanced throttling policy](https://apim.docs.wso2.com/en/latest/learn/rate-limiting/adding-new-throttling-policies/#adding-a-new-advanced-throttling-policy)
+1.  Conditional throttle policies are an advance set of API and Resource level policies. Thefore you need to define the required policy to apply in the OpenAPI definition. To do that, define `x-wso2-throttling-tier` extension with the Advance Throttle Policy name you defined in API Manager in above step. This extension can be defined in both API and Resource levels.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:217]:
```suggestion
1.  Conditional throttle policies are an advanced set of API and Resource level policies. Therefore you need to define the required policy to apply in the OpenAPI definition. To do that, define `x-wso2-throttling-tier` extension with the Advance Throttle Policy name you defined in API Manager in the above step. This extension can be defined in both API and Resource levels.
```

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Review by shaniR — CHANGES_REQUESTED** [review]

**Review by shaniR — APPROVED** [review]

**Pull request merged**